datainterface/fastener.py

```diff
@@ -0,0 +1,564 @@
+# fastener.py
+
+import os
+from glob import glob
+from uuid import UUID
+import json
+import datetime
+from abc import ABC, abstractmethod
+from fractions import Fraction
+from decimal import Decimal
```

Comment:
Are these modules all automatically included in `pip3`? If not, can we set up a python environment that can be activated with these modules in it so that we don't run into any module-not-found errors. Evan's got one set up for the sim, perhaps we could move it to the root and have it apply to all Python code in the repo.

Reply:
Yes, I believe all these modules are included in `pip3`.